---
语言: cpp
语法类型: 中级语法
---
`try-catch` 结构构成 C++ 异常处理，表示异常处理块中的捕获块，实现处理错误状态的代码与正常代码分开

```cpp
try {
    // do something
} catch (const std::exception &e) {
    // do something
} catch (const std::exception&) {
    // do something
} catch (...) {
    // do something
}
```

> [!warning] 异常处理不应替代程序中应有的数据检查和验证，异常处理带来的系统开销相当大

当我们不关心具体的异常值时，异常变量可省略（示例第二个 `catch` 块）

当我们使用 `...` 作为捕获变量时，表示该块可以捕获任意类型的异常

使用 `throw 异常对象;` 可以抛出一个异常

`catch` 中可直接使用 `throw` 重新抛出未捕获的异常

内存申请相关的异常为 `new` 头文件中的 `bad_alloc&` 类型

使用 [[异常列表|throws]] 标注一个函数可能抛出的异常，但用的很少

使用 [[noexcept]] 标注一个函数一定不会产生异常

> [!note] MFC 异常为 `CException*` 类型，并使用 `Delete()` 方法释放
